Receive RESOURCE_EXHAUSTED status when running test with latest stratum-bfrt container with Tofino Model

Test: fabric-tna, profile: fabric
Stratum container: stratum-bfrt:latest-9.2.0 (20201204-9.2.0)

### Notes

- I add an additional flag to disable the ONLP and fix the chassis config (see https://github.com/stratum/fabric-tna/pull/134)
- I tried with both python2 and python3 based PTF test, both of them return `RESOURCE_EXHAUSTED` error
- I tried this on both my laptop (Macbook Pro) and a VM(KVM based, Ubuntu 20.04, 32C32G), and both of them return the same error.

### Full log from the command line when running the test

```
$ ./ptf/run/tm/run fabric
*** Testing profile 'fabric'...
*** Detected BF SDE version 9.2.0 (tofino-model), required 9.2.0...
*** Using P4 compiler output in /home/vagrant/fabric-tna/ptf/run/tm/../../../p4src/build/fabric/sde_9_2_0...
*** Starting tofino-model-20543 (from opennetworking/bf-sde:9.2.0-tm)...
367ded03b53311a3512b219d1667cc8267d0fc52e3e8bd319d9727749397efe3
*** Starting stratum-bf-20543 (from registry.aetherproject.org/tost/stratum-bfrt:latest-9.2.0)...
35c26eafbdfa3e01f2e93d053687ca4e75ddcd27ee2232dfad836d7ca46f7e51
*** Wait for stratum-bf-20543 to start up (port 28000)...
wait-for-it.sh: waiting 600 seconds for localhost:28000
wait-for-it.sh: localhost:28000 is available after 7 seconds
*** Starting tester-20543...
************************************************
STARTING PTF TESTS...
************************************************
python3 -u ptf_runner.py --port-map port_map.veth.json --ptf-dir fabric.ptf --cpu-port 320 --device-id 1 --grpc-addr "127.0.0.1:28000" --p4info /p4c-out/p4info.txt --tofino-pipeline-config /p4c-out/pipeline_config.pb.bin all ^spgw ^int ^bng ^dth ^xconnect ^p4rt 
INFO:PTF runner:Sending P4 config
ERROR:PTF runner:Error during SetForwardingPipelineConfig
ERROR:PTF runner:<_InactiveRpcError of RPC that terminated with:
        status = StatusCode.RESOURCE_EXHAUSTED
        details = "to-be-sent initial metadata size exceeds peer limit"
        debug_error_string = "{"created":"@1607159813.061940445","description":"Error received from peer ipv4:127.0.0.1:28000","file":"src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"to-be-sent initial metadata size exceeds peer limit","grpc_status":8}"
>
../../tests/ptf/Makefile.profiles:12: recipe for target 'fabric' failed
make: *** [fabric] Error 2
************************************************
SOME PTF TESTS FAILED :(
************************************************
*** Stopping stratum-bf-20543...
*** Stopping tofino-model-20543...
```

### Stratum log

```
I20201205 09:16:51.511572    12 main_bfrt.cc:78] switchd started successfully
W20201205 09:16:51.511695    12 credentials_manager.cc:45] Using insecure server credentials
I20201205 09:16:51.512060    12 hal.cc:128] Setting up HAL in COLDBOOT mode...
I20201205 09:16:51.512143    12 config_monitoring_service.cc:94] Pushing the saved chassis config read from /workdir/chassis_config.pb.txt...
E20201205 09:16:51.518813    12 phal.cc:96] No phal_config_file specified and no switch configurator found! This is probably not what you want. Did you forget to specify any '--define phal_with_*=true' Bazel flags?
E1205 09:16:51.520086942      12 server_chttp2.cc:40]        {"created":"@1607159811.520058232","description":"Only 1 addresses added out of total 2 resolved","file":"external/com_github_grpc_grpc/src/core/ext/transport/chttp2/server/chttp2_server.cc","file_line":406,"referenced_errors":[{"created":"@1607159811.520053954","description":"Address family not supported by protocol","errno":97,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":403,"os_error":"Address family not supported by protocol","syscall":"socket","target_address":"[::1]:28003"}]}
I20201205 09:16:51.520193    12 attribute_database.cc:213] PhalDB service is listening to localhost:28003...
I20201205 09:16:51.520233    12 bf_chassis_manager.cc:989] Port status notification callback registered successfully
I20201205 09:16:51.520893    12 bf_chassis_manager.cc:87] Adding port 1 in node 1 (SDK Port 0).
I20201205 09:16:51.520936    12 bf_chassis_manager.cc:116] Enabling port 1 in node 1 (SDK Port 0).
I20201205 09:16:51.521044    12 bf_chassis_manager.cc:87] Adding port 2 in node 1 (SDK Port 4).
I20201205 09:16:51.521064    12 bf_chassis_manager.cc:116] Enabling port 2 in node 1 (SDK Port 4).
I20201205 09:16:51.521138    12 bf_chassis_manager.cc:87] Adding port 3 in node 1 (SDK Port 8).
I20201205 09:16:51.521157    12 bf_chassis_manager.cc:116] Enabling port 3 in node 1 (SDK Port 8).
I20201205 09:16:51.521219    12 bf_chassis_manager.cc:87] Adding port 4 in node 1 (SDK Port 12).
I20201205 09:16:51.521237    12 bf_chassis_manager.cc:116] Enabling port 4 in node 1 (SDK Port 12).
I20201205 09:16:51.521296    12 bf_chassis_manager.cc:87] Adding port 5 in node 1 (SDK Port 16).
I20201205 09:16:51.521314    12 bf_chassis_manager.cc:116] Enabling port 5 in node 1 (SDK Port 16).
I20201205 09:16:51.521677    12 bf_chassis_manager.cc:87] Adding port 6 in node 1 (SDK Port 20).
I20201205 09:16:51.521698    12 bf_chassis_manager.cc:116] Enabling port 6 in node 1 (SDK Port 20).
I20201205 09:16:51.521788    12 bf_chassis_manager.cc:87] Adding port 7 in node 1 (SDK Port 24).
I20201205 09:16:51.521807    12 bf_chassis_manager.cc:116] Enabling port 7 in node 1 (SDK Port 24).
I20201205 09:16:51.521979    12 bfrt_switch.cc:60] Chassis config pushed successfully.
I20201205 09:16:51.529151    12 p4_service.cc:121] Pushing the saved forwarding pipeline configs read from /dev/null...
W20201205 09:16:51.529284    12 p4_service.cc:142] Empty forwarding pipeline configs file: /dev/null.
E1205 09:16:51.529554857      12 server_chttp2.cc:40]        {"created":"@1607159811.529527306","description":"Only 1 addresses added out of total 2 resolved","file":"external/com_github_grpc_grpc/src/core/ext/transport/chttp2/server/chttp2_server.cc","file_line":406,"referenced_errors":[{"created":"@1607159811.529525119","description":"Address family not supported by protocol","errno":97,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":403,"os_error":"Address family not supported by protocol","syscall":"socket","target_address":"[::1]:28000"}]}
E20201205 09:16:51.529723    12 hal.cc:221] Stratum external facing services are listening to 0.0.0.0:28000, localhost:28000...
E20201205 09:16:51.530290    12 hal.cc:363] StratumErrorSpace::ERR_INTERNAL: Failed to check in with procmon: failed to connect to all addresses
E20201205 09:16:51.530553    12 hal.cc:231] Error when checking in with procmon: Failed to check in with procmon: failed to connect to all addresses.
I20201205 09:16:53.055428    42 p4_service.cc:759] Controller (connection_id: 1, election_id: 1, uri: ipv4:127.0.0.1:42174) is connected as MASTER for node (aka device) with ID 1.
I20201205 09:16:53.062918    42 p4_service.cc:786] Controller (connection_id: 1, election_id: 1, uri: ipv4:127.0.0.1:42174) which was MASTER for node (aka device) with ID 1 is disconnected. The node is now orphan :(
```

--- a/stratum/hal/bin/barefoot/README.run.md
+++ b/stratum/hal/bin/barefoot/README.run.md
@@ -382,7 +382,8 @@ ERROR:PTF runner:<_InactiveRpcError of RPC that terminated with:
 
 This error originates from the gRPC layer and can occur when the pipeline is
 particularly large and does not fit in the [maximum receive message size](https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_builder.html#ab5c8a420f2acfc6fcea2f2210e9d426e).
-Although we set a reasonable default, the value can be adjusted with the `-grpc_max_recv_msg_size` flag.
+Although we set a reasonable default, the value can be adjusted with Stratum's
+`-grpc_max_recv_msg_size` flag.
 
 ### TNA P4 programs on Stratum-bf / PI Node
 

--- a/stratum/hal/bin/barefoot/README.run.md
+++ b/stratum/hal/bin/barefoot/README.run.md
@@ -364,3 +364,22 @@ this particular switch model. As a workaround, [BSP mode](#Running-with-BSP-or-o
 which bypasses ONLP, is available.
 
 [start-stratum-container-sh]: https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/docker/start-stratum-container.sh
+
+### RESOURCE_EXHAUSTED when pushing pipeline
+
+Stratum rejects a SetForwardingPipelineConfig request with a RESOURCE_EXHAUSTED
+gRPC error, like this:
+
+```
+INFO:PTF runner:Sending P4 config
+ERROR:PTF runner:Error during SetForwardingPipelineConfig
+ERROR:PTF runner:<_InactiveRpcError of RPC that terminated with:
+        status = StatusCode.RESOURCE_EXHAUSTED
+        details = "to-be-sent initial metadata size exceeds peer limit"
+        debug_error_string = "{"created":"@1607159813.061940445","description":"Error received from peer ipv4:127.0.0.1:28000","file":"src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"to-be-sent initial metadata size exceeds peer limit","grpc_status":8}"
+>
+```
+
+This error originates from the gRPC layer and can occur when the pipeline is
+particularly large and does not fit in the [maximum receive message size](https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_builder.html#ab5c8a420f2acfc6fcea2f2210e9d426e).
+Although we set a reasonable default, the value can be adjusted with the `-grpc_max_recv_msg_size` flag.

